packages/react-data-grid/src/cells/headerCells/SortableHeaderCell.js

```diff
@@ -22,13 +23,13 @@ class SortableHeaderCell extends React.Component {
     case null:
     case undefined:
     case DEFINE_SORT.NONE:
-      direction = DEFINE_SORT.ASC;
+      direction = this.props.sortDescendingFirst ? DEFINE_SORT.DESC : DEFINE_SORT.ASC;
```

Comment:
This can be cleaned up little using destructuring 
```
const { sortDirection, sortDescendingFirst } = this.props;
...
switch (sortDirection) {
...
case DEFINE_SORT.NONE:
direction = sortDescendingFirst ? DEFINE_SORT.DESC : DEFINE_SORT.ASC;
```